### PR TITLE
Feature/#2867 create orders page

### PR DESCRIPTION
**Pull request by AlekOK**
I created a new page for orders. There I added mock data.

### CODE REVIEW

**Review by Anastasiia-Hevchuk — APPROVED** [review]

**Review by YulianDev — APPROVED** [review]

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  <div class="mainHeader">Ваші замовлення</div>
+  <div *ngIf="isAnyOrders" class="ifEmpty"><span>Ви ще не зробили замовлень у нас.</span><a href="#">Зробити замовлення</a></div>
+  <div *ngIf="!isAnyOrders" class="cards">
+    <div class="headerList">
+      <div>№ замовлення</div>
+      <div>Статус</div>
+      <div>Сума оплати</div>
+    </div>
+    <div *ngFor="let order of orders" class="card">
+      <div class="emptyCard">
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:11]: use snake case for class names

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html [hunk 2]
```diff
@@ -0,0 +1,112 @@
+<div class="wrapper">
+  <div class="mainHeader">Ваші замовлення</div>
+  <div *ngIf="isAnyOrders" class="ifEmpty"><span>Ви ще не зробили замовлень у нас.</span><a href="#">Зробити замовлення</a></div>
+  <div *ngIf="!isAnyOrders" class="cards">
+    <div class="headerList">
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:5]: use snake case for class names

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      <div>Сума оплати</div>
+    </div>
+    <div *ngFor="let order of orders" class="card">
+      <div class="emptyCard">
+        <img *ngIf="order.extend" src="assets/img/ubs-admin-orders/arrow_down.svg" alt="arrow down" (click)="changeCard(order.id)" />
+        <img *ngIf="!order.extend" src="assets/img/ubs-admin-orders/arrow_up.svg" alt="arrow up" (click)="changeCard(order.id)" />
+        <div>{{ order.order }}</div>
+        <div>{{ order.status }}</div>
+        <div class="orderSum">{{ order.sum }}</div>
+        <div class="btnsGroup">
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:17]: use snake case for class names
and fix this for all clasess

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+          <div *ngIf="order.status === 'Сформовано'" class="btnGridCancel"><button class="btnCancel">Скасувати</button></div>
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:18]: don't use UA lang for codding

Comment (JimHolman1199) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:18]: move button for next line

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            <td>-100 грн</td>
+          </tr>
+          <tr class="sumToPay">
+            <td colspan="4">Сума до оплати:</td>
+            <td>350 грн</td>
+          </tr>
+        </table>
+      </div>
+      <div *ngIf="order.extend" class="orderDetails">
+        <ol style="list-style-type: none">
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:85]: move this to class
also maybe use Bootstrap

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  isAnyOrders: boolean;
+
+  constructor(private adminOrdersService: AdminOrdersService) {}
+
+  ngOnInit() {
+    this.adminOrdersService
+      .getOrders()
+      .pipe(takeUntil(this.destroy))
+      .subscribe((item) => {
+        console.log(item);
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts:23]: console.log ?

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        this.orders = item;
+      });
+  }
+
+  changeCard(id) {
+    this.orders.map((el) => {
+      el.id === id && el.extend ? el.extend = false : el.id === id ? el.extend = true : el.extend = el.extend;
+    });
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts:31]: this will return an array of undefined ? no ?

Comment (JimHolman1199) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts:31]: you should return el

File: src/app/ubs-admin/services/admin-orders.service.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdminOrdersService {
+  backend = 'assets/orders.json';
```
Comment (JimHolman1199) [inline src/app/ubs-admin/services/admin-orders.service.ts:8]: move this to mocks folder

**Review by tkhristina — APPROVED** [review]

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html (hunk 1 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:11]: Done

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html (hunk 2 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:5]: Done

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html (hunk 3 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:17]: Done

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html (hunk 5 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:85]: Moved this property to the scss file.

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts (hunk 6 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts:23]: Removed

File: src/app/ubs-admin/services/admin-orders.service.ts (hunk 8 above)
Comment (AlekOK) [reply src/app/ubs-admin/services/admin-orders.service.ts:8]: backend = 'assets/mocks/orders/orders.json';

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts (hunk 7 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts:31]: No. This map return the same array, but with changed property "extend". 
By the way that property is mine. So, in the future, when we get real data, the property and method will change.

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html (hunk 4 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:18]: moved

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    </div>
+    <div *ngFor="let order of orders" class="card">
+      <div class="empty_card">
+        <img *ngIf="order.extend" src="assets/img/ubs-admin-orders/arrow_down.svg" alt="arrow down" (click)="changeCard(order.id)" />
+        <img *ngIf="!order.extend" src="assets/img/ubs-admin-orders/arrow_up.svg" alt="arrow up" (click)="changeCard(order.id)" />
+        <div>{{ order.order }}</div>
+        <div>{{ order.status }}</div>
+        <div class="order_sum">{{ order.sum }}</div>
+        <div class="btns_group">
+          <div *ngIf="order.status === 'Сформовано'" class="btn_grid_cancel">
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:18]: don't use UA lang for coding

Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:18]: I added a new property in data object - "order_status". This property can have the values: "formed", "done"...

**Review by dyadyushko — APPROVED** [review]

**Review by SofiiaBaziv — APPROVED** [review]

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.scss [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+  background-color: var(--after-primary-light-grey);
+}
+
+.main_header {
+  margin: 0 0 51px 44px;
+  font-style: normal;
+  font-weight: 700;
+  font-size: 32px;
+  line-height: 40px;
+  letter-spacing: 0.002em;
```
Comment (Edw1x) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.scss:17]: 0.002 * 16(wrapper's font-size) = 0.032px < 1px

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+      .getOrders()
+      .pipe(takeUntil(this.destroy))
+      .subscribe((item) => {
+        this.orders = item;
+      });
+  }
+
+  changeCard(id) {
+    this.orders.map((el) => {
+      el.id === id && el.extend ? (el.extend = false) : el.id === id ? (el.extend = true) : (el.extend = el.extend);
```
Comment (Edw1x) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts:29]: We should change it to if-else statement
Hard to read :/

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.scss [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+  grid-template-rows: 28px auto;
+  grid-template-columns: 670px;
+  margin-left: 56px;
+  color: var(--ubs-secondary-grey);
+
+  .header_details {
+    font-size: 20px;
+    line-height: 28px;
+    font-weight: bold;
+    letter-spacing: 0.002em;
```
Comment (Edw1x) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.scss:127]: same

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+          <tr class="sum_to_pay">
+            <td colspan="4">Сума до оплати:</td>
+            <td>350 грн</td>
+          </tr>
+        </table>
+      </div>
+      <div *ngIf="order.extend" class="order_details">
+        <ol>
+          <li class="bold_text">Доставка з Еко магазину</li>
+          <li>Номери ваших замовлень:<span class="bold_text">&nbsp;3507777,</span>&nbsp;<span class="boldText">45678907777</span></li>
```
Comment (Edw1x) [inline src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:94]: We need to break this line into new lines

Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.html:94]: Done

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts (hunk 11 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.ts:29]: As for me this is better... 
Also we will change this logic when we will recive real data.

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.scss (hunk 10 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.scss:17]: Resolved

File: src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.scss (hunk 12 above)
Comment (AlekOK) [reply src/app/ubs-admin/components/ubs-admin-orders/ubs-admin-orders.component.scss:127]: Resolved

**Review by dyadyushko — APPROVED** [review]

**Review by Edw1x — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

**Pull request merged**